src/deepsparse/server/config.py

```diff
@@ -0,0 +1,206 @@
+    )
+
+
+class ServerConfig(BaseModel):
+    """
+    A configuration for serving models in the DeepSparse inference server
+    """
+
+    models: List[ServeModelConfig] = Field(
+        default=[],
+        description=(
+            "The models to serve in the server defined by the additional arguments"
+        ),
+    )
+    workers: str = Field(
+        default=max(1, cpu_architecture().num_available_physical_cores // 2),
+        description=(
+            "The number of maximum workers to use for processing pipeline requests. "
+            "Defaults to the number of physical cores on the device."
+        ),
+    )
+
+
+@lru_cache()
+def server_config_from_env(env_key: str = ENV_DEEPSPARSE_SERVER_CONFIG):
+    """
+    Load a server configuration from the targeted environment variable given by env_key.
+
+    :param env_key: the environment variable to load the configuration from.
+        Defaults to ENV_DEEPSPARSE_SERVER_CONFIG
+    :return: the loaded configuration file
+    """
+    config_file = os.environ[env_key]
+
+    if not config_file:
+        raise ValueError(
+            "environment variable for deepsparse server config not found at "
+            f"{env_key}"
+        )
+
+    if config_file.startswith(ENV_SINGLE_PREFIX):
+        config_dict = json.loads(config_file.replace(ENV_SINGLE_PREFIX, ""))
+        config = ServerConfig()
+        config.models.append(
+            ServeModelConfig(
+                task=config_dict["task"],
+                model_path=config_dict["model_path"],
+                batch_size=config_dict["batch_size"],
+            )
+        )
+    else:
+        with open(config_file) as file:
+            config_dict = yaml.safe_load(file.read())
+        config_dict["models"] = (
+            [ServeModelConfig(**model) for model in config_dict["models"]]
+            if "models" in config_dict
+            else []
+        )
+        config = ServerConfig(**config_dict)
+
+    if len(config.models) == 0:
+        raise ValueError(
+            "There must be at least one model to serve in the configuration "
+            "for the deepsparse inference server"
+        )
+
+    return config
+
+
+def server_config_to_env(
+    config_file: str,
+    task: str,
+    model_path: str,
+    batch_size: int,
+    env_key: str = ENV_DEEPSPARSE_SERVER_CONFIG,
+):
+    """
+    Put a server configuration in an environment variable given by env_key.
+    If config_file is given, ignores task, model_path, and batch_size.
+    Otherwise, creates a configuration file from task, model_path, and batch_size
+    for serving a single model.
+
+    :param config_file: the path to the config file to store in the environment
+    :param task: the task the model_path is serving such as question_answering.
+        If config_file is supplied, this is ignored.
+    :param model_path: the path to a model.onnx file, a model folder containing
+        the model.onnx and supporting files, or a SparseZoo model stub.
```

Comment:
have you tested the model folder method? this isn't supported in the default deepsparse flow i.e. compile_model

Reply:
yes, this is tested for transformer pipeline flows